src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackDecoder.cs

```diff
@@ -192,6 +201,15 @@ public void Decode(ReadOnlySpan<byte> data, HeaderCallback onHeader, object onHe
                         }
                         else if ((b & DynamicTableSizeUpdateMask) == DynamicTableSizeUpdateRepresentation)
                         {
+                            // https://tools.ietf.org/html/rfc7541#section-4.2
+                            // This dynamic table size
+                            // update MUST occur at the beginning of the first header block
+                            // following the change to the dynamic table size.
+                            if (_headersObserved)
+                            {
+                                throw new HPackDecodingException("Dynamic table size update not at beginning of header block.");
```

Comment:
These error messages should be in the .resx

Reply:
Okay, just learned that when I start my own review it doesn't make comments visible until I submit the review. Yea, I'm aware -- I intend to ask @wfurt for guidance re: l10n.